test/dotnet-add-p2p.Tests/GivenDotnetAddP2P.cs

```diff
@@ -0,0 +1,90 @@
+        private static readonly string Lib2Path = Path.Combine("Lib2", "Lib2.csproj");
+        private static readonly string Lib3Path = Path.Combine("Lib3", "Lib3.csproj");
+        private static readonly string Lib4Path = Path.Combine("Lib4", "Lib4.csproj");
+
+        private string Setup([System.Runtime.CompilerServices.CallerMemberName] string callingMethod = nameof(Setup))
+        {
+            const string TestGroup = "NonRestoredTestProjects";
+            const string ProjectName = "DotnetAddP2PProjects";
+            return GetTestGroupTestAssetsManager(TestGroup).CreateTestInstance(ProjectName, callingMethod: callingMethod).Path;
+        }
+
+        [Theory]
+        [InlineData("--help")]
+        [InlineData("-h")]
+        public void WhenHelpOptionIsPassedItPrintsHelp(string helpArg)
+        {
+            var cmd = new AddP2PCommand().Execute(helpArg);
+            cmd.Should().Pass();
+            cmd.StdOut.Should().Contain("Usage");
+        }
+
+        [Theory]
+        [InlineData("idontexist")]
+        [InlineData("Lib1")] // folder name
+        [InlineData("Broken.csproj")] // exists but broken
+        public void WhenInvalidProjectIsPassedItFails(string projName)
+        {
+            string testRoot = Setup();
+            new AddP2PCommand()
+                    .WithWorkingDirectory(testRoot)
+                    .Execute($"{projName} -r {Lib1Path}")
+                .ExitCode.Should().NotBe(0);
+        }
+
+        [Fact]
+        public void WhenConditionAlreadyExistsOnItemItDoesntDuplicate()
+        {
+            string testRoot = Setup();
+            CountOccurrances(File.ReadAllText(Path.Combine(testRoot, Lib2Path)), "Lib4.csproj").Should().Be(2);
+            new AddP2PCommand()
+                    .WithWorkingDirectory(testRoot)
+                    .Execute($"{Lib2Path} -r {Lib4Path} -f net451")
+                .Should().Pass();
+            new AddP2PCommand()
+                    .WithWorkingDirectory(testRoot)
+                    .Execute($"{Lib2Path} -r {Lib4Path} -f netcoreapp1.0")
+                .Should().Pass();
+
+            string lib2FullPath = Path.Combine(testRoot, Lib2Path);
+            CountOccurrances(File.ReadAllText(lib2FullPath), "Lib4.csproj").Should().Be(2);
+
+            new RestoreCommand()
+                .WithWorkingDirectory(testRoot)
+                .Execute($"\"{lib2FullPath}\"")
+                .Should().Pass();
+
+            new BuildCommand()
+                .WithWorkingDirectory(testRoot)
+                .Execute($"\"{lib2FullPath}\"")
+                .Should().Pass();
+        }
+
+        private static int CountOccurrances(string s, string pattern)
+        {
+            int ret = 0;
+            for (int i = s.IndexOf(pattern); i != -1; i = s.IndexOf(pattern, i + 1))
+            {
+                ret++;
+            }
+
+            return ret;
+        }
+    }
+}
```

Comment:
Sure. You and @blackdwarf should scope the feature, but let's track scenarios that are out-of-scope. I think globs are common enough that we may want to provide some special handling e.g. `globs not supported right now` though that may not be needed.. you guys will decide.